scene/resources/packed_scene.cpp

```diff
@@ -943,6 +944,19 @@ Error SceneState::_parse_node(Node *p_owner, Node *p_node, int p_parent_idx, Has
 		}
 	}
 
+	if (!p_node->get_scene_file_path().is_empty() && p_owner != p_node && !p_owner->is_editable_instance(p_node)) {
+		TypedArray<Node> children = p_node->get_exposed_nodes();
```

Comment:
With the current implementation of `get_exposed_nodes()`, this method will search the whole sub-tree for each node in the scene 🙃
You should cache exposed nodes, similar to how unique nodes are cached. They can be stored in a HashSet or LocalVector, it doesn't need to be HashMap.

Reply:
So something akin to this?

```cpp
TypedArray<Node> Node::get_exposed_nodes(bool p_recursive) const {
	ERR_THREAD_GUARD_V(TypedArray<Node>())
	TypedArray<Node> ret;
	for (Node* node : data.owned_exposed_nodes) {
		if (!node->get_scene_file_path().is_empty()) {
			continue;
		}
		if (node->is_exposed_in_owner()) {
			ret.append(node);
		}
		if (p_recursive) {
			ret.append_array(node->get_exposed_nodes(true));
		}
	}

	return ret;
}
```

where node has something like :
```cpp
	HashSet<Node *> owned_exposed_nodes;
	void _release_exposed_in_owner();
	void _acquire_exposed_in_owner();
```